src/NHapi.Base/Parser/DefaultXMLParser.cs

```diff
@@ -97,71 +99,59 @@ public static new void Main(string[] args)
+            {
+                messageName = messageName?.Replace("+", string.Empty);
+            }
+
+            XmlDocument doc;
             try
             {
                 doc = new XmlDocument();
-                var root = doc.CreateElement(messageName);
+                var root = doc.CreateElement(messageName, NameSpace);
                 doc.AppendChild(root);
             }
             catch (Exception e)
             {
                 throw new HL7Exception(
-                    "Can't create XML document - " + e.GetType().FullName,
+                    $"Can't create XML document - {e.GetType().FullName}",
                     ErrorCode.APPLICATION_INTERNAL_ERROR,
                     e);
             }
 
-            Encode(source, (XmlElement)doc.DocumentElement);
+            Encode(source, doc.DocumentElement, parserOptions);
             return doc;
         }
 
-        /// <summary> <p>Creates and populates a Message object from an XML Document that contains an XML-encoded HL7 message.</p>
-        /// <p>The easiest way to implement this method for a particular message structure is as follows:
-        /// <ol><li>Create an instance of the Message type you are going to handle with your subclass
-        /// of XMLParser</li>
-        /// <li>Go through the given Document and find the Elements that represent the top level of
-        /// each message segment. </li>
-        /// <li>For each of these segments, call <code>parse(Segment segmentObject, Element segmentElement)</code>,
-        /// providing the appropriate Segment from your Message object, and the corresponding Element.</li></ol>
-        /// At the end of this process, your Message object should be populated with data from the XML
-        /// Document.</p>
-        /// </summary>
-        /// <throws>  HL7Exception if the message is not correctly formatted. </throws>
-        /// <throws>  EncodingNotSupportedException if the message encoded. </throws>
-        /// <summary>     is not supported by this parser.
-        /// </summary>
+        /// <inheritdoc/>
         public override IMessage ParseDocument(XmlDocument xmlMessage, string version, ParserOptions parserOptions)
         {
-            if (parserOptions is null)
+            if (xmlMessage is null)
             {
-                throw new ArgumentNullException(nameof(parserOptions));
+                throw new ArgumentNullException(nameof(xmlMessage));
             }
 
-            var messageName = xmlMessage.DocumentElement.Name;
+            parserOptions ??= DefaultParserOptions;
+
+            AssertNamespaceUri(xmlMessage.DocumentElement?.NamespaceURI);
+
+            var messageName = xmlMessage.DocumentElement?.LocalName;
             var message = InstantiateMessage(messageName, version, true);
             Parse(message, xmlMessage.DocumentElement, parserOptions);
             return message;
         }
 
+        // TODO: should this be public?
```

Comment:
Should this TODO (and others in this pull request) be tagged with an issue number so they don't get lost/forgotten?

Reply:
This is just something I need to look into, since I think default access modifiers are different in java compared to C# and I'm not sure `nHapi` has correctly captured this - I have created an issue for this #399.